src/dex/swaap-v2/swaap-v2.ts

```diff
@@ -578,33 +593,62 @@ export class SwaapV2 extends SimpleExchange implements IDex<SwaapV2Data> {
           this.logger.warn(
             `${this.dexKey}-${this.network}: protocol is restricted`,
           );
-          await this.restrict();
+          const poolIdentifier = getPoolIdentifier(
+            this.dexKey,
+            normalizedSrcToken.address,
+            normalizedDestToken.address,
+          );
+          var message = 'Unknown error';
+          if (e instanceof Error) {
+            message = `${e.name}: ${e.message}`;
+          }
+          await this.restrictPool(message, poolIdentifier);
         }
       }
 
       throw e;
     }
   }
 
-  async restrict(ttl: number = SWAAP_RESTRICT_TTL_S): Promise<boolean> {
-    await this.dexHelper.cache.setex(
-      this.dexKey,
-      this.network,
-      SWAAP_RESTRICTED_CACHE_KEY,
-      ttl,
-      'true',
+  async restrictPool(message: string, poolIdentifier: string): Promise<void> {
+    this.logger.warn(
+      `${this.dexKey}-${this.network}: ${poolIdentifier} was restricted for ${SWAAP_POOL_RESTRICT_TTL_S} sec. due to fails`,
     );
-    return true;
-  }
 
-  async isRestricted(): Promise<boolean> {
-    const result = await this.dexHelper.cache.get(
-      this.dexKey,
-      this.network,
-      SWAAP_RESTRICTED_CACHE_KEY,
+    // We use timestamp for creation date to later discern if it already expired or not
+    await this.dexHelper.cache.hset(
+      this.runtimeMMsRestrictHashMapKey,
+      poolIdentifier,
+      Date.now().toString(),
     );
 
-    return result === 'true';
+    this.rateFetcher
+      .notify(SWAAP_BANNED_CODE, message, this.getNotifyReqParams())
+      .then((answer: SwaapV2NotificationResponse) => {
```

Comment:
@JackieJoo Could we use async/await ?

Reply:
we'd need to wrap it in a separate try-catch, seems easier to just use `.then().catch()`